src/modules/job-exec/exec.c

```diff
@@ -308,8 +300,43 @@ static void exec_exit (struct jobinfo *job)
     job->data = NULL;
 }
 
+/*  Configure the exec module.
+ *  Read the default job shell path from config. Allow override on cmdline
+ */
+static int exec_config (flux_t *h, int argc, char **argv)
+{
+    flux_conf_error_t err;
+
+    /*  Set default job shell path from builtin configuration,
+     *   allow override via configuration, then cmdline.
+     */
+    default_job_shell = flux_conf_builtin_get ("shell_path", FLUX_CONF_AUTO);
+
+
+    /*  Check configuration for exec.job-shell */
+    if (flux_conf_unpack (flux_get_conf (h, NULL),
```

Comment:
The first call to `flux_get_conf()` reads and parses TOML then creates the "config object", while subsequent calls just access it from the handle aux cache.  The first call (wherever that is) should check for errors.

Reply:
Ah, ok. Does every handle reread all config files? I had assumed that the broker read the config and checked for parse errors and made other validity checks, and therefore the config was known to be valid in modules. Sorry about the bad assumption on my part.

Reply:
I guess the broker isn't going to re-read config files on module reload anyway, so this makes sense now. Thanks!

Reply:
It's not ideal and could use some thought on how to improve, but that's the current state of things.  Each `flux_t` handle is operating independently with respect to config.

Reply:
This config scheme is actually a bit tricky to test. To test that a module fails when reading a bad config file,  you have to first start an instance with a good set of config files, then replace the config files with one that will cause TOML parse errors. Then reload the module. Unless currently there is a way to set the config file dir at `flux module reload` time?

Reply:
Agreed, @dongahn ran into a similar issue when adding config support to qmanager - see #2759.  I didn't take any action on it though, sorry.

Maybe it's OK to not be too rigorous in the config testing here until we've improved the world?

Reply:
It didn't end up being too bad:
```
test_expect_success 'job-exec: bad TOML config causes module failure' '
	#  Need to first start an instance with good config files, then
        #   replace with bad config files, and finally reload module
	name=badtoml &&
	mkdir ${name}.d &&
	touch ${name}.d/exec.toml &&
	( export FLUX_CONF_DIR=${name}.d &&
	  test_must_fail flux start -s1 sh -c "\
	    echo \[exec >${name}.d/exec.toml && flux module reload job-exec" \
	    >${name}.log 2>&1
	) &&
        grep "config file error: ${name}.d/exec.toml" ${name}.log

'
```